### PR TITLE
Undocumented/storybook 8

### PR DESCRIPTION
**Pull request by WULCAN**
## Description
This PR updates Storybook so that it can be started and used again.

## Notes to reviewer

Please see the commit message for details.

I did a manual upgrade to 7.0, following 
https://github.com/storybookjs/storybook/blob/next/MIGRATION.md because I found that easier to debug than upgrading directly to Storybook 8. I did not adapt to all deprecations at first.

Our addons weren't compatible with Storybook 7 so I replaced them.

I tested a prior version of this upgrade to 7.0 on d75089814ea4c786490ad934d3155db5be639616: the revision just before the upgrade to Next 13 and 14. In secret, I was hoping that 7.0 would be enough to get it running again, but alas, my hopes were in vain.

Luckily, the automatic upgrade to Storybook 8 worked pretty well and only a few relatively simple fixes had to be applied upon it. Some of them were handling the deprecations from 7 that I tried to skip before.

See also undocumented/upgrade-storybook that @ziggabyte has been working on.

One of the last rounds of testing showed that we have some new unmocked requests but. See the console in developer tools. Those will most likely be easy to fix but let's do that in a later pull request. This one is big enough already and I as Storybook actually starts with it, I consider it an improvement over main, even if it isn't perfect.

I also found a couple of small bugs while testing on 7.0 that I opted to not fix on this branch:
* src/zui/ZUIPersonLink/index.stories.tsx wraps in Typography, which is not allowed
* src/zui/ZUISuffixedNumber/index.stories.tsx sends args to the first parameter of bind, but that parameter is for this, so the args are not used by Storybook and nothing is rendered

There's probably a lot more in https://github.com/storybookjs/storybook/blob/next/MIGRATION.md that we should review, but let's defer that too to a later pull request.

This is my first pull with code to request to app.zetkin.org =)

### CODE REVIEW

**Review by richardolsson — COMMENTED** [review]
Awesome work @WULCAN! I have looked at the code changes and I have tried the Storybook build and dev modes. All looks good to me! I created issue #2051 with some ideas for next steps (based on conversations with several of you recently).

I'd love to hear @ziggabyte's thoughts as well, and will leave it to you to merge into main and then back into the work that you've been doing.

File: .storybook/preview.js [hunk 1]
```diff
@@ -25,37 +27,55 @@ const I18nProvider = (props) => {
...
+  } else {
+    return Promise.reject({ error: 'unmocked request', path, init });
+  }
+}
+
+class MockApiClient extends FetchApiClient {
+  constructor() {
+    super(mockFetch);
+  }
+}
```
Comment (ziggabyte) [inline .storybook/preview.js:50]: I like this! :D

File: .storybook/preview.js [hunk 2]
```diff
@@ -25,37 +27,55 @@ const I18nProvider = (props) => {
...
+    </ThemeProvider>
+  ),
+  (story) => {
+    const store = createStore();
+    const router = useRouter();
+    const env = new Environment(store, new MockApiClient(), router);
+    return (
+      <ReduxProvider store={store}>
+        <EnvProvider env={env}>{story()}</EnvProvider>
+      </ReduxProvider>
```
Comment (ziggabyte) [inline .storybook/preview.js:65]: I added all the providers but I guess these are the ones that are actually needed! Nice to see your solution :D

File: src/features/calendar/components/EventCluster/index.stories.tsx [hunk 3]
```diff
@@ -9,9 +9,9 @@ import Single from './Single';
 export default {
   component: Single,
   title: 'Calendar',
-} as ComponentMeta<typeof Single>;
+} as Meta<typeof Single>;
 
-const SingleEvent: ComponentStory<typeof Single> = (args) => (
+const SingleEvent: StoryFn<typeof Single> = (args) => (
```
Comment (ziggabyte) [inline src/features/calendar/components/EventCluster/index.stories.tsx:14]: Ah it was ok to update the existing stories like this! Nice!

**Review by ziggabyte — APPROVED** [review]
So happy that you got this to work, my thoughts when I did my thing was "ok someone with more expierience and patience could probably make this work" and YES you did it!!!

File: .storybook/preview.js (hunk 1 above)
Comment (WULCAN) [reply .storybook/preview.js:50]: Yay! It's definitely more verbose than the configuration format for storybook-addon-mock but on the other hand it requires no knowledge of any addon to understand exactly what it does.

File: .storybook/preview.js (hunk 2 above)
Comment (WULCAN) [reply .storybook/preview.js:65]: Yeah, I don't know which is better of `Providers` or my mix of decorators. On `main` we have `ReduxProvider` and `I18nProvider` as separate decorators and I just built on that. I didn't know about `Providers` until I saw your branch.

File: src/features/calendar/components/EventCluster/index.stories.tsx (hunk 3 above)
Comment (WULCAN) [reply src/features/calendar/components/EventCluster/index.stories.tsx:14]: Yeah, that's how I interpret https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#decoratorfn-story-componentstory-componentstoryobj-componentstoryfn-and-componentmeta-typescript-types and it seems to work.

**Pull request merged**